packages/namekit-react/src/components/Identity.tsx

```diff
@@ -0,0 +1,292 @@
+    <div className={`namekit-identity namekit-address ${className}`} {...props}>
+      {address}
+    </div>
+  );
+};
+
+const NameGuardShield = ({ className, ...props }: SubComponentProps) => {
+  const { identityData, returnNameGuardReport, loadingState } = useIdentity();
+
+  if (
+    !returnNameGuardReport ||
+    loadingState !== "success" ||
+    !identityData?.nameguard_report
+  ) {
+    return null;
+  }
+
+  return (
+    <div className={`namekit-nameguard-shield ${className}`} {...props}>
+      <div className="namekit-nameguard-rating">
+        Rating: {identityData.nameguard_report.rating}
+      </div>
+      <div className="namekit-nameguard-risk-count">
+        Risks: {identityData.nameguard_report.risk_count}
+      </div>
+    </div>
+  );
+};
+
+const Followers = ({ className, ...props }: SubComponentProps) => {
+  const { followersCount, loadingState } = useIdentity();
+
+  if (loadingState === "loading") {
+    return (
+      <div className={`namekit-followers-skeleton ${className}`} {...props}>
+        Loading followers...
+      </div>
+    );
+  }
+
+  if (followersCount === undefined) {
+    return (
+      <div className={`namekit-followers-loading ${className}`} {...props}>
+        Fetching followers...
+      </div>
+    );
+  }
+
+  return (
+    <div
+      className={`namekit-identity namekit-followers ${className}`}
+      {...props}
+    >
+      {followersCount} followers
+    </div>
+  );
+};
+
+const ENSLogo = () => (
+  <svg
+    fill="none"
+    height="16"
+    viewBox="0 0 202 231"
+    width="14"
+    xmlns="http://www.w3.org/2000/svg"
+  >
+    <g fill="#0080bc">
+      <path d="m98.3592 2.80337-63.5239 104.52363c-.4982.82-1.6556.911-2.2736.178-5.5924-6.641-26.42692-34.89-.6463-60.6377 23.5249-23.4947 53.4891-40.24601 64.5942-46.035595 1.2599-.656858 2.587.758365 1.8496 1.971665z" />
+      <path d="m94.8459 230.385c1.2678.888 2.8299-.626 1.9802-1.918-14.1887-21.581-61.3548-93.386-67.8702-104.165-6.4264-10.632-19.06614-28.301-20.12056-43.4178-.10524-1.5091-2.19202-1.8155-2.71696-.3963-.8466 2.2888-1.74793 5.0206-2.58796 8.1413-10.60469 39.3938 4.79656 81.1968 38.24488 104.6088l53.0706 37.148z" />
+      <path d="m103.571 228.526 63.524-104.523c.498-.82 1.656-.911 2.274-.178 5.592 6.64 26.427 34.89.646 60.638-23.525 23.494-53.489 40.246-64.594 46.035-1.26.657-2.587-.758-1.85-1.972z" />
+      <path d="m107.154.930762c-1.268-.8873666-2.83.625938-1.98 1.918258 14.189 21.58108 61.355 93.38638 67.87 104.16498 6.427 10.632 19.066 28.301 20.121 43.418.105 1.509 2.192 1.815 2.717.396.846-2.289 1.748-5.02 2.588-8.141 10.604-39.394-4.797-81.1965-38.245-104.609z" />
+    </g>
+  </svg>
+);
+
+const ENSProfileLink = ({ className, ...props }: SubComponentProps) => {
+  const { identityData, loadingState } = useIdentity();
+
+  if (loadingState !== "success" || !identityData?.display_name) {
+    return null;
+  }
+
+  return (
+    <a
+      href={`https://app.ens.domains/${identityData.display_name}`}
```

Comment:
We need some strategy for people who use NameKit React to customize the link target for viewing a detailed profile. This needs to support a variety of cases, including calling a function to a traditional href, customizing opening in a new tab or not, etc..

Additionally, the way this link is built may vary. For example, in some apps the link to a profile is based on an address. In other apps, the link is based on an ENS Name (where possible) and only uses a link with an address as a fallback if the address has no primary name.

For example, consider an app such as EFP. When someone clicks on a profile link on their app, they don't want it to open to the official ENS App. They want it to open to the profile page on their own app.

I worked on some ideas for how a goal like this in the following PR. Open to other suggested strategies for achieving this dynamic logic: Check the logic around the file packages/nameguard-react/src/utils/openreport.ts it might give some inspiration? https://github.com/namehash/namekit/pull/282/files#diff-4b7ca1b4f908b4cf07782ef287606daa30f6c17f8181e6d7a3ab82eb09137dbd

Reply:
@lightwalker-eth this is what I have in mind that I would like to experiment this week...

## 1. Standard implementation
This shows the default link:

```tsx
<Identity.Root address={address}>
  <Identity.ENSProfileLink />
</Identity.Root>
```

## 2. Custom NameKit Config

```tsx
const efpConfig = {
  profileLinks: {
    getProfileTarget: ({ address, ensName }) => 
      `/profile/${ensName || address}`,
    openInNewTab: false,
    // Optionally use Next.js Link or other routing component
    LinkComponent: NextLink,
  }
};

<NameKitConfigProvider config={efpConfig}>
  <Identity.Root address={address}>
    <Identity.ENSProfileLink />
  </Identity.Root>
</NameKitConfigProvider>
```

Optionally, a click handler similar to what you suggested:

```tsx
const handlerConfig = {
  profileLinks: {
    getProfileTarget: ({ address, ensName }) => 
      (e: React.MouseEvent) => {
        e.preventDefault();
        // Custom handling logic
        openProfileModal(address, ensName);
      }
  }
};
```

This allows us to move towards the "React Query" context provider for data too.

Reply:
@notrab Hey, thanks for putting this together and sharing it 🙌 🚀 

A few suggestions:

1. `getProfileTarget` am I correct to understand this is supposed to return a string representing a URL? If so, suggest renaming this to `getProfileURL`.
   1. I also see your separate click handler example, but confused how everything fits together here. Because here the `getProfileTarget` doesn't return a string, it returns a function that takes a `React.MouseEvent` param.
2. `openInNewTab` and `LinkComponent` I like the general direction you're working towards here for supporting more customizability. But maybe we use a different strategy? What do you think of modifying the approach there to instead have a function called `getProfileLink` that takes in a `url` (as might be generated by `getProfileURL` and then returns a DOM element tree for turning that `url` into the HTML for a link.
   1. This strategy not only would make customizing link targets / link components no problem, but I believe it should also give a solution for the "click handler" case.
   2. Maybe instead of `getProfileLink` taking as input a `url` as a param, it should instead take the same param type as `getProfileTarget` / `getProfileURL`. Then, if `getProfileLink` wants to build a link with a URL in it, it can call `getProfileURL` itself to do that. Alternatively, if `getProfileLink` wants to do some javascript-based "click handler" it can then more easily define that logic if the param passed to `getProfileLink` were an address or an ensName.
3. For now, maybe we exclusively build support for getting the profiles using address, not using ensName. I'm worried supporting both will slow us down right now. It's definitely nice to ultimately add support for both in the future, but suggest we come back to that later. For example, in the future we might define some special interface for what gets passed into `Identity.Root` where it is either a name or an address and not both. This same interface could then be passed around to other downstream components such as the callback functions defined in the config, etc.. But suggest we keep things more simple for now and only focus on identities that are anchored in an address and not an ensName.
4. Suggest that we define some default handlers for the profile link configs. The default should result in opening to view the profile of the provided address on the ens.domains website in a new tab. Ex: https://app.ens.domains/0xf81bc66316A3f2A60Adc258F97F61dFcBdd23Bb1
5. It seems nice to make it so that `Identity.ENSProfileLink` just defines a link, not the anchor elements for the link`. As I understand we should allow any children to be passed into `Identity.ENSProfileLink` which would then become the anchor for the link. As I understand, just doing `<Identity.ENSProfileLink />` should be similar to doing `<a />` in that there shouldn't be anything for a user to see or interact with until you define the children beneath that.
6. Whatever interface we decide on for setting the config for profile links, it might be nice to allow a custom config for a specific `Identity.ENSProfileLink` instance to be passed in. By default people wouldn't do that, so then it would fallback to the default config set in `NameKitConfigProvider`.
7. Perhaps good to rename `Identity.ENSProfileLink` to just `Identity.ProfileLink`. I like making "ENS" implied rather than explicit. For example, use `name` instead of `EnsName`, etc.. If we don't do this we'll have "ENS" too much all over our code.

Appreciate your advice :+1:

Reply:
@lightwalker-eth I've applied most of this feedback in a new commit. I agree with it all.

I also updated the stories to match the new requirements too. I hope I didn't miss anything, it took some refactoring/rethinking to make it all work better 🙏🏻

Reply:
@lightwalker-eth I made some changes based on what you said about the idea of a class for the `ProfileLinkGenerator`.

```ts
export class ProfileLinkGenerator {
  private baseURL: string;
  private name: string;

  constructor(name: string, baseURL: string) {
    this.name = name;
    this.baseURL = baseURL;
  }

  getName(): string {
    return this.name;
  }

  getProfileURL(address: string): string {
    return `${this.baseURL}${address}`;
  }
}
```

This is basic for now, but acts as a way to set a base URL, and we could even export some default ones, such as:

```ts
export const ENSProfileLink = new ProfileLinkGenerator(
  "ENS",
  "https://app.ens.domains/",
);
```

Then this can be used wherever:

```ts
interface ProfileLinkProps {
  config?: ProfileLinkGenerator;
  className?: string;
  children?: React.ReactNode;
  onClick?: (e: React.MouseEvent) => void;
}

const ProfileLink: React.FC<ProfileLinkProps> = ({ config, children, onClick }) => {
  const identity = useIdentity();
  const nameKitConfig = useNameKitConfig();

  const linkConfig = config || nameKitConfig.profileLinks?.[0] || DEFAULT_PROFILE_LINKS[0];

  if (!identity) {
    console.warn("ProfileLink used outside of Identity context");
    return null;
  }

  const url = linkConfig.getProfileURL(identity.address);

  return (
    <a
      href={url}
      target="_blank"
      rel="noopener noreferrer"
      className="namekit-profile-link"
      onClick={onClick}
    >
      {children || linkConfig.getName()}
    </a>
  );
};
```

This also allows ProfileLinks to be controlled individually, or part of the context provider down. Giving more flexibility.